pandas/core/common.py

```diff
@@ -2510,6 +2510,8 @@ def is_floating_dtype(arr_or_dtype):
 
 
 def is_bool_dtype(arr_or_dtype):
+    if arr_or_dtype is NotImplemented:
```

Comment:
So conceptually though, the code in the wrapper got a result, its trying to check if this result is a bool type ... the check is crashing on this NULL type of exception. So this is where its gets tricky conceptually, is NULL part of this functions domain or not?